packages/hash/frontend/src/pages/[account-slug]/types/entity-type/entity-type-selector.tsx

```diff
@@ -0,0 +1,236 @@
+import { EntityType } from "@blockprotocol/type-system-web";
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon, TextField } from "@hashintel/hash-design-system";
+import {
+  Autocomplete,
+  Box,
+  outlinedInputClasses,
+  PopperProps,
+  Typography,
+} from "@mui/material";
+import { FunctionComponent, useMemo, useRef, useState } from "react";
+import { ArrowUpRightIcon } from "../../../../shared/icons/svg";
+import {
+  OntologyChip,
+  parseUriForOntologyChip,
+} from "../../shared/ontology-chip";
+import {
+  TypeListSelectorDropdown,
+  TypeListSelectorDropdownContext,
+  TYPE_SELECTOR_HEIGHT,
+} from "./type-list-selector-dropdown";
+import { useEntityTypes } from "./use-entity-types";
+
+export const EntityTypeSelector: FunctionComponent<{
```

Comment:
Much of this is duplicative of the property type selector. Is there no way to generalise it? Particularly the style parts of it. A quick check of the diff between this and the property type selector suggests they're very similar.

<img width="812" alt="image" src="https://user-images.githubusercontent.com/719814/204430194-bc30ef05-9a95-40c1-95df-28c719c4b589.png">

Reply:
Implemented here https://github.com/hashintel/hash/pull/1492/commits/32fc65022334ea474f7c634c04685d31acd9a6cd